### ISSUE TITLE
Promise Support

### ISSUE THREAD

**Issue by ccollie** [jpwilliams/microboot#27]
Given that they're everywhere, I suggest support be added for promises. Quick suggestion  follows (loader.js)

```
var debugLoad = require('debug')('microboot:load');
var microloader = require('microloader');

module.exports = load;

function load (paths) {
  let callbacks = [];

  microloader(paths, {
    absolute: true
  }).forEach(function (file) {
    debugLoad('- loading "' + file + '"...');

    let ret = require(file);

    if (typeof ret === 'function') {
      debugLoad('- loaded "' + file + '" and added' + (ret.name ? (' "' + ret.name + '"') : '') + ' to list');

      ret = wrapFunction(ret);
      callbacks.push({
        func: ret,
        path: file
      });

    } else {
      debugLoad('- loaded "' + ret + '"');
    }
  });

  debugLoad('- loaded ' + callbacks.length + ' functions');

  return callbacks
}

// wrap single argument accepting boot function to handle
// promises
function wrapFunction(fn) {
  if (fn.length > 1) {
    return fn;
  }
  return function (arg, done) {
    let res;
    try {
      res = fn(arg);
    } catch (err) {
      return done(err);
    }
    if (res && typeof(res.then) === 'function') {
      res.then(() => done()).catch(done);
    } else {
      done();
    }
  }
}

```

**Comment by jpwilliams** [comment]
This is addressed in [v3.0.0](https://github.com/jpwilliams/microboot/releases/tag/3.0.0). 👍

**Comment by ccollie** [comment]
Many thanks
